### PR TITLE
Github Actions: Add Phpstan

### PR DESCRIPTION
**Pull request by sukhwinder33445**

### CODE REVIEW

File: phpstan.neon
Comment (nilmerg) [inline phpstan.neon:1]: Same here, is there something that should be part of the baseline?

File: src/Cron.php [hunk 1]
```diff
@@ -162,6 +162,7 @@ public static function isValid(string $expression): bool
 
     public static function fromJson(string $json): Frequency
     {
+        /** @var array<string, string> $data */
```
Comment (nilmerg) [inline src/Cron.php:165]: Why is this needed? Is this about that json_decode may return null? Then we should handle it accordingly and not mute the error with such meaningless inline doctype.

File: src/OneOff.php [hunk 2]
```diff
@@ -45,6 +45,7 @@ public function getEnd(): ?DateTimeInterface
 
     public static function fromJson(string $json): Frequency
     {
+        /** @var string $data */
```
Comment (nilmerg) [inline src/OneOff.php:48]: same here

File: src/RRule.php [hunk 3]
```diff
@@ -316,6 +323,8 @@ public function __call(string $methodName, array $args)
             );
         }
 
-        return call_user_func_array([$this->rrule, $methodName], $args);
+        return call_user_func_array(function (...$args) use ($methodName) {
+            return $this->rrule->$methodName(...$args);
+        }, $args);
```
Comment (nilmerg) [inline src/RRule.php:328]: This looks pointless. What has been fixed here?

**Review by nilmerg — CHANGES_REQUESTED** [review]

File: src/RRule.php (hunk 3 above)
Comment (yhabteab) [reply src/RRule.php:328]:
```php
Line   RRule.php                                                        
 ------ ----------------------------------------------------------------- 
  326    Parameter #1 $callback of function call_user_func_array expects  
         callable(): mixed, array{Recurr\Rule, string} given.    
```

File: src/OneOff.php (hunk 2 above)
Comment (yhabteab) [reply src/OneOff.php:48]:
```php
  Line   OneOff.php                                                            
 ------ ---------------------------------------------------------------------- 
  50     Parameter #1 $datetime of class DateTime constructor expects string,  
         mixed given.  
```

File: src/Cron.php (hunk 1 above)
Comment (yhabteab) [reply src/Cron.php:165]:
```php
Line   Cron.php                                                              
 ------ ---------------------------------------------------------------------- 
  167    Cannot access offset 'expression' on mixed.                           
  168    Cannot access offset 'start' on mixed.                                
  169    Cannot access offset 'start' on mixed.
...
```

Comment (nilmerg) [reply src/Cron.php:165]: Then throw an error if the result is not an array.

File: src/OneOff.php (hunk 2 above)
Comment (nilmerg) [reply src/OneOff.php:48]: Same here, throw an error if it's not string.

File: src/RRule.php (hunk 3 above)
Comment (yhabteab) [reply src/RRule.php:328]: https://github.com/phpstan/phpstan/issues/1242#issuecomment-439716243

Comment (yhabteab) [reply src/RRule.php:328]: https://github.com/phpstan/phpstan/issues/1210#issuecomment-404482691

Comment (nilmerg) [reply src/RRule.php:328]: Oi. Then ignore the error. This abomination isn't going to be merged.

**Comment by nilmerg** [comment]
@cla-bot check
